### PR TITLE
Create AWS resource `aws.eks.addon`

### PR DESCRIPTION
**Pull request by mm-weber**
Trying to fix https://github.com/mondoohq/cnquery/issues/4222

### CODE REVIEW

File: providers/aws/resources/aws.lr [hunk 1]
```diff
@@ -3073,6 +3073,31 @@ private aws.eks.nodegroup @defaults("name scalingConfig.DesiredSize diskSize sta
   autoscalingGroups() []aws.autoscaling.group
 }
 
+// Amazon EKS add-on
+private aws.eks.addon @defaults("addonName clusterName addonVersion status") {
+  // The name of the add-on.
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // The name of the add-on
```

File: providers/aws/resources/aws.lr [hunk 2]
```diff
@@ -3073,6 +3073,31 @@ private aws.eks.nodegroup @defaults("name scalingConfig.DesiredSize diskSize sta
...
+  name string
+  // The Amazon Resource Name (ARN) of the add-on.
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // The Amazon Resource Name (ARN) of the add-on
```

File: providers/aws/resources/aws.lr [hunk 3]
```diff
@@ -3073,6 +3073,31 @@ private aws.eks.nodegroup @defaults("name scalingConfig.DesiredSize diskSize sta
...
+  arn() string
+  // The status of the add-on
+  status string
+  // The version of the add-on
+  addonVersion string
+  // The Unix epoch timestamp at object creation.
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // The Unix epoch timestamp at object creation
```

File: providers/aws/resources/aws.lr [hunk 4]
```diff
@@ -3073,6 +3073,31 @@ private aws.eks.nodegroup @defaults("name scalingConfig.DesiredSize diskSize sta
...
+  createdAt() time
+  // The Unix epoch timestamp for the last modification to the object.
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // The Unix epoch timestamp for the last modification to the object
```

File: providers/aws/resources/aws.lr [hunk 5]
```diff
@@ -3073,6 +3073,31 @@ private aws.eks.nodegroup @defaults("name scalingConfig.DesiredSize diskSize sta
...
+  modifiedAt() time
+  // Tags for the EKS node group
+  tags()  map[string]string
+  // The publisher of the add-on.
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // The publisher of the add-on
```

File: providers/aws/resources/aws.lr [hunk 6]
```diff
@@ -3073,6 +3073,31 @@ private aws.eks.nodegroup @defaults("name scalingConfig.DesiredSize diskSize sta
...
+  publisher string
+  // The owner of the add-on.
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // The owner of the add-on
```

File: providers/aws/resources/aws.lr [hunk 7]
```diff
@@ -3073,6 +3073,31 @@ private aws.eks.nodegroup @defaults("name scalingConfig.DesiredSize diskSize sta
...
+  owner string
+  // The configuration values that you provided.
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // The configuration values that you provided
```

File: providers/aws/resources/aws.lr [hunk 8]
```diff
@@ -3103,4 +3128,6 @@ private aws.eks.cluster @defaults("arn version status") {
   createdAt time
   // List of EKS node groups
   nodeGroups() []aws.eks.nodegroup
+  // List of EKS addons
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // List of EKS add-ons
```

**Review by misterpantz — COMMENTED** [review]
Hi @mm-weber! These descriptions are very clear. :) I just have suggestions for consistency.

File: providers/aws/resources/aws.lr.manifest.yaml [hunk 9]
```diff
@@ -1513,8 +1513,27 @@ resources:
...
+      configurationValues: {}
+      createdAt: {}
+      modifiedAt: {}
+      name: {}
+      owner: {}
+      publisher: {}
+      status: {}
+      tags: {}
+    is_private: true
+    min_mondoo_version: latest
```
Comment (tas50) [inline providers/aws/resources/aws.lr.manifest.yaml]:
```suggestion
    min_mondoo_version: 9.0.0
```

File: providers/aws/resources/aws.lr [hunk 10]
```diff
@@ -3073,6 +3073,31 @@ private aws.eks.nodegroup @defaults("name scalingConfig.DesiredSize diskSize sta
...
+  tags()  map[string]string
+  // The publisher of the add-on
+  publisher() string
+  // The owner of the add-on
+  owner() string
+  // The configuration values that you provided
+  configurationValues() string
+}
+
+
```
Comment (tas50) [inline providers/aws/resources/aws.lr]:
```suggestion
```

File: providers/aws/resources/aws.lr.manifest.yaml [hunk 11]
```diff
@@ -1513,8 +1513,27 @@ resources:
...
+      tags: {}
+    is_private: true
+    min_mondoo_version: 9.0.0
+    platform:
+      name:
+      - aws
   aws.eks.cluster:
     fields:
+      addons:
+        min_mondoo_version: latest
```
Comment (tas50) [inline providers/aws/resources/aws.lr.manifest.yaml]:
```suggestion
        min_mondoo_version: 9.0.0
```

**Comment by tas50** [comment]
@mm-weber something is causing the same add-on to be listed multiple times:

```
  addons: [
    0: aws.eks.addon name="coredns" addonVersion="v1.11.1-eksbuild.4" status="ACTIVE"
    1: aws.eks.addon name="coredns" addonVersion="v1.11.1-eksbuild.4" status="ACTIVE"
    2: aws.eks.addon name="coredns" addonVersion="v1.11.1-eksbuild.4" status="ACTIVE"
    3: aws.eks.addon name="coredns" addonVersion="v1.11.1-eksbuild.4" status="ACTIVE"
  ]
```

**Review by tas50 — APPROVED** [review]
Works great now

```
  addons: [
    0: aws.eks.addon name="coredns" addonVersion="v1.11.1-eksbuild.4" status="ACTIVE"
    1: aws.eks.addon name="eks-pod-identity-agent" addonVersion="v1.2.0-eksbuild.1" status="ACTIVE"
    2: aws.eks.addon name="kube-proxy" addonVersion="v1.29.3-eksbuild.2" status="ACTIVE"
    3: aws.eks.addon name="vpc-cni" addonVersion="v1.18.0-eksbuild.1" status="ACTIVE"
  ]
```

**Pull request merged**